src/remote.ts

```diff
@@ -528,6 +528,34 @@ export class Remote {
     }
 
     await sshConfig.update(sshValues, sshConfigOverrides)
+
+    // A user can provide a "Host *" entry in their SSH config to add options
+    // to all hosts. We need to ensure that the options we set are not
+    // overridden by the user's config.
+    const computedProperties = computeSSHProperties(hostName, sshConfig.getRaw())
+    const keysToMatch: Array<keyof SSHValues> = ["ProxyCommand", "UserKnownHostsFile", "StrictHostKeyChecking"]
+    for (let i = 0; i < keysToMatch.length; i++) {
+      const key = keysToMatch[i]
+      if (computedProperties[key] === sshValues[key]) {
+        continue
+      }
+
+      const result = await this.vscodeProposed.window.showErrorMessage(
+        "Unsupported SSH Config Option",
+        {
+          useCustom: true,
+          modal: true,
+          detail: `Your SSH config is overriding the "${key}=${computedProperties[key]}" option for the "${hostName}" host. Please remove it and try again!`,
```

Comment:
We could add "overriding the key=correct with key=wrong ..." to help trace the issue?

Reply:
Good point!